auth-backends/README.adoc

```diff
@@ -811,4 +811,157 @@ It means that the role `node_all` is not recognized. It is because it is not a r
-`rudder.auth.oauth2.provider.okta.roles.mapping.entitlements.role-oidc-a=access_to_node`
+`rudder.auth.oauth2.provider.okta.roles.mapping.entitlements.role-oidc-a=access_to_node`.
+
+
+=== OAuth2 tokens : JWT and opaque bearer tokens
+
+OAuth2 tokens are used to grant access to resources, using authentication tokens, serving the purpose of authenticating machines, automated systems, etc.
+There are two types of tokens implemented as authentication backends in Rudder: JWT (JSON Web Tokens) and opaque bearer tokens.
+
+JWT tokens contain user information and permissions in a JSON format. They can be verified using a signature, allowing for local validation without contacting the authorization server.
+
+Opaque bearer tokens do not contain user information. They serve as a reference to session or authorization data stored on the server. To validate these tokens, the resource server must contact the authorization server to obtain user details.
+
+Both are implemented with configuration that is similar to the provisioning of users with OIDC, except that tokens are involved :
+
+* for JWT, Rudder can locally validate the token's signature, ensuring that the token has not been tampered with and that it is issued by a trusted source. This local validation reduces the need for frequent communication with the authorization server, improving performance and reliability.
+* for opaque bearer tokens, Rudder relies on the authorization server to validate the token. This means that each request with an opaque token requires a call to the authorization server to verify the user's identity and permissions. While this can introduce some latency, it ensures that the most up-to-date user information is used when resolving token authentication.
+
+Tenants and roles mapping can be configured for both :
+
+* for JWT, the token has custom attributes
+* for opaque bearer token, the token is bound with an API account in Rudder, which can have ACLs and tenants
+
+WARNING: In Rudder, you should use only one of both tokens, provided that they are configured correctly (see below for example configurations)
+
+==== Obtaining and using JWT token in Rudder
+
+First you will need to obtain a valid JWT token from your IdP, for example from Okta :
+
+```
+curl --request POST 'https://xxxx.okta.com/oauth2/xxxxaudiencexxxx/v1/token' \
+--header 'Accept: application/json' \
+--header 'Content-Type: application/x-www-form-urlencoded' \
+--header 'Authorization: Basic xxx' \
+--data 'grant_type=client_credentials'
+```
+
+It will return an access token in the payload that should directly be usable with the Rudder REST API as a `Bearer` token i.e. with the header `Authorization: Bearer the_access_token`.
+
+Below is an example configuration for handling JWT tokens in Rudder from an external OAuth2 provider `okta`. You will need to replace `okta` with your actual provider.
+
```

Comment:
Please had "copy the following snippet in /opt/rudder/et/rudder-web.properties.d/oauth2-api-jwt.properties" so that people don't wonder where to put them and - Hopefully  - we ease future debug session by knowing where the config will be. Plus it's a good practice to put that in a dedicated file